NetTopologySuite.IO/NetTopologySuite.IO/IO/Streams/FileStreamProvider.cs

```diff
@@ -19,10 +21,10 @@ public FileStreamProvider(string kind, string path, bool validatePath = false)
             if (path == null)
                 throw new ArgumentNullException("path");
 
-#if NET40
+#if HAS_SYSTEM_STRING_ISNULLORWHITESPACE
             if (string.IsNullOrWhiteSpace(path))
 #else
-            if (string.IsNullOrEmpty(path))
+            if (string.IsNullOrEmpty(path) || path.All(Char.IsWhiteSpace))

```

Comment:
Is there are more efficient helper method we can use instead of this LINQ crap, preferably without keeping the gap in functionality?